app/serializers/concerns/api/v2/csv/csv_serializer.rb

```diff
@@ -0,0 +1,58 @@
+module Api
+  module V2
+    module Csv
+      module CsvSerializer
+        extend ActiveSupport::Concern
+
+        def initialize(serializables, options = {})
+          @serializables = serializables
+          @options = options
+        end
+
+        def serializable_array
+          @serializables.each_with_object([header_row]) do |serializable, acc|
+            data_row = column_options.map do |column_option|
+              if column_option[:value_block].present?
+                column_option[:value_block].call(serializable)
+              else
+                serializable.public_send(column_option[:column])
+              end
+            end
+
+            acc << data_row
+          end
+        end
+
+        def serialized_csv
+          CSV.generate do |csv|
+            serializable_array.each do |row|
+              csv << row
+            end
+          end
+        end
+
+        delegate :column_options, to: :class
+
+        private
+
+        def header_row
+          column_options.map do |column_option|
+            column_option[:column_name]
+          end
+        end
+
+        module ClassMethods
+          cattr_accessor :column_options
```

Comment:
I think this will mean column options end up shared across all classes including the module - I think you need class level instance accessors instead

```ruby
module CsvSerializer
  extend ActiveSupport::Concern
  
  included do
    class << self
      attr_accessor :column_options
    end
  end
end
```

Reply:
Ignore this, better option in comment below